src/i18n.tsx

```diff
@@ -1,34 +1,77 @@
 import React from 'react'
 import { i18n } from '@lingui/core'
 import { I18nProvider } from '@lingui/react'
-import { detect, fromUrl } from '@lingui/detect-locale'
-import { ReactNode, useEffect } from 'react'
-
-export const locales = ['en', 'pseudo-en', 'de', 'es-AR', 'es-US', 'it-IT', 'iw', 'ro', 'ru', 'vi', 'zh-CN', 'zh-TW']
-export const defaultLocale = 'en'
-
-const getDetectedLocale = () => {
-  const detected =
-    detect(
-      fromUrl('lang'), // helps local development
-      defaultLocale
-    ) ?? defaultLocale
-  return locales.includes(detected) ? detected : defaultLocale
+import { ReactNode } from 'react'
+import { useDispatch } from 'react-redux'
+import useParsedQueryString from 'hooks/useParsedQueryString'
+import { useEffect } from '@storybook/addons'
+import { AppDispatch } from 'state'
+import { useLocale } from 'state/user/hooks'
```

Comment:
this is a circular import (even though not a circular dependency) and might make sense to put the locale in a constants file